src/TTimer.cpp

```diff
@@ -71,6 +71,14 @@ TTimer::~TTimer()
     mpQTimer->stop();
     if (mpHost) {
         mpHost->getTimerUnit()->unregisterTimer(this);
+
+        if (isTemporary()) {
+            if (mScript.isEmpty()) {
+                mpHost->mLuaInterpreter.delete_luafunction(this);
+            } else {
+                mpHost->mLuaInterpreter.delete_luafunction(QStringLiteral("Timer%1").arg(mName));
```

Comment:
💭 Is this something that only applies to `TTimer` objects or is it likely to be something that we should be doing for other item types?

Reply:
Yes, applies to all other types as well!